distributed/client.py

```diff
@@ -1294,34 +1297,65 @@ async def _update_scheduler_info(self):
         except OSError:
             logger.debug("Not able to query scheduler for identity")
 
-    async def _wait_for_workers(self, n_workers=0, timeout=None):
+    async def _wait_for_workers(
+        self,
+        n_workers: int = 0,
+        timeout: int | None = None,
```

Comment:
```suggestion
        timeout: Optional[int]  = None,
```
(type-based `|` is Python >= 3.9, needs a `from typing import Optional`)

Reply:
No need since `from __future__ import annotations` (ignore me).